Tests/Marketplace/marketplace_services.py

```diff
@@ -1567,6 +1651,179 @@ def prepare_release_notes(self, index_folder_path, build_number,
         finally:
             return task_status, not_updated_build
 
+    def filter_changelog_entries(self, changelog_entry: dict, version: str, modified_files_data: dict,
+                                 marketplace: str):
+        """
+        Filters the changelog entries by the entities that are given from id-set.
+        This is to avoid RN entries/changes/messages that are not relevant to the current marketplace.
+
+        The filter is done in two parts:
+        1. Filter the entry by marketplace intended tags.
+        2. Filter by the entity display name if it doesn't exist in id-set.
+
+        If there are not entries after filtering then the pack will be skipped and not be uploaded.
+
+        Args:
+            changelog_entry: The version changelog object.
+            version: The changelog's version.
+            modified_files_data: The data from id_set for the modified entities.
+
+        Returns:
+            (dict) The filtered changelog entry.
+            (bool) Whether the pack is not updated because the entries are not relevant to the current marketplace.
+        """
+
+        release_notes = self.filter_release_notes_by_tags(changelog_entry.get(Changelog.RELEASE_NOTES), marketplace)
+
+        # Convert the RN entries to a Dict
+        release_notes_dict, _ = merge_version_blocks(pack_versions_dict={version: release_notes},
```

Comment:
I would create a wrapper function `get_merged_release_notes_dict` that does this line (and explain on its docstring what "merged release notes" are).